server/machine_interfaces/haas.py

```diff
@@ -0,0 +1,192 @@
+
+        # set machine config
+        self._machine_config = machine_config
+
+        # set serial port name (i.e. COM3)
+        self._serial_port = serial.Serial(port = serial_port, baudrate = 9600, bytesize = serial.SEVENBITS, xonxoff = True, timeout = 5)
+
+        # Attempt to open a serial port for communication with the Haas
+        try:
+            self._serial_port.open()
+        except serial.SerialException:
+            if self._serial_port.is_open:
+                try:
+                    print("Port was open, attempting to close and reopen")
+                    self._serial_port.close()
+                    time.sleep(2)
+                    self._serial_port.open()
+                except:
+                    print("Failed to close port")
+                    event.clear()
+            else:
+                print("Failed to connect to serial port, either in use or does not exist")
+                event.clear() 
+
+        # set machine name
+        self._machine_name = self._machine_config["Haas CNC"]
+
+        # set up publish address
+        self._publish_address = f"tcp://127.0.0.1:{publish_port}"
+
+        # set up publish socket
+        self._context = zmq.Context()
+        self._publish_socket = self._context.socket(zmq.PUB)
+        retry(
+            self._publish_socket.connect,
+            self._publish_address,
+            handled_exceptions=zmq.error.ZMQError
+        )
+
+        # initialize thread
+        Thread.__init__(self)
+        self._stopped = False
+
+    def run(self):
+        # loop until thread is stopped
+        while not self._stopped:
+            # get machine data
+            machine_data = self._poll_machine()
+
+            # publish machine data
+            self._publish_data(self._publish_socket, self._machine_name, machine_data)
+
+        # close ZMQ socket and context
+        self._close_zmq()
+
+    def stop(self):
+        self._stopped = True
+
+    # Redefine abstract method to poll a machine for data
+    def _poll_machine(self):
+
+        # Helper function to get write macros to Haas requesting specific data
+        def get_data(ser, constant, output):
+            try:
+                ser.write(constant)
+                start_timer = time.time() # start a timer
+                while True:
+                    output = ser.readline().decode("utf-8")
+                    if len(output) > 4:
+                        break
+                    delta = time.time() - start_timer
+                    if delta >= 3: # wait 3 seconds to see if data arrives
+                        output = '0' # no data recieved
+                        break
+                output = output.split(",")[2].strip()[0:8]
+            except:
+                output = '0'
+            return output
+
+        print("Begin data collection:")
+
+        # Begin fetching data from Haas
+        try:
+            # Coolant Level Data
+            coolant_lvl = ''
```

Comment:
this, and all of the other `<var> = ''` statements do nothing, since you're setting the variable on the next line